lib/rpc/SingleJsonRpcProvider.ts

```diff
@@ -472,14 +474,41 @@ export class SingleJsonRpcProvider extends StaticJsonRpcProvider {
     )
   }
 
-  override call(
+  override call(transaction: Deferrable<TransactionRequest>, blockTag?: BlockTag | Promise<BlockTag>): Promise<string> {
+    return this.wrappedFunctionCall(CallType.NORMAL, 'call', super.call.bind(this), transaction, blockTag)
+  }
+
+  override send(method: string, params: Array<any>): Promise<any> {
+    return this.wrappedFunctionCall(CallType.NORMAL, 'send', super.send.bind(this), method, params)
+  }
+
+  ///////////////////// Begin of special functions /////////////////////
+
+  // Wrap another layer only for the sake of ease unit testing.
+  // We will test this API to represent the tests of other similar implemented APIs.
+  private _getBlockNumber(): Promise<number> {
+    return super.getBlockNumber()
+  }
+
+  private getBlockNumber_EvaluateHealthiness(): Promise<number> {
+    return this.wrappedFunctionCall(CallType.HEALTH_CHECK, 'getBlockNumber', this._getBlockNumber.bind(this))
```

Comment:
i'm curious about this pattern: is the binding necessary since `_getBlockNumber` is already a method of this class ?

Reply:
This is necessary as we are forwarding the function to another function `wrappedFunctionCall`. Without this bind, it will use the wrong `this` context.

Reply:
I tried it to verify that it is indeed needed..